packages/grid/src/Grid.tsx

```diff
@@ -175,190 +167,221 @@ const Grid = ({
+  const [rightFixedWidth, setRightFixedWidth] = useState(0);
 
-      if (newFullWidth < width) {
-        const lastColIdx = searchLastVisible(newColumns, newColumns.length);
-        newColumns = setIn(
-          newColumns,
-          newColumns[lastColIdx].width + (width - newFullWidth),
-          [String(lastColIdx), 'width'],
-        );
 
-        newFullWidth = newColumns.reduce(
-          (acc, { width: colWidth }) => (acc + colWidth),
-          0,
-        );
-      }
-      fullWidthRef.current = newFullWidth;
-      setMappedColumns(newColumns);
-      onChangeColumns(newColumns);
-    },
-    [mappedColumns, onChangeColumns, width],
-  );
+  useEffect(() => {
+    setLeftMappedColumns(mappedColumns.filter(({ fixedPosition }: IColumn) => fixedPosition === 'left'));
 
+    setCenterMappedColumns(mappedColumns.filter(({ fixedPosition }: IColumn) => !fixedPosition));
 
-  const handleChangeMoving = useCallback(
-    (newColumns) => {
-      setMappedColumns(newColumns);
-      onChangeColumns(newColumns);
-    },
-    [onChangeColumns],
-  );
+    setRightMappedColumns(mappedColumns.filter(({ fixedPosition }: IColumn) => fixedPosition === 'right'));
+  }, [mappedColumns]);
 
   useEffect(() => {
-    if (gridRef.current) gridRef.current.recomputeGridSize();
-    if (cacheRef.current) cacheRef.current.clearAll();
-  }, [mappedColumns, mappedItems]);
+    setLeftFixedWidth(
+      leftMappedColumns
+        .filter(({ visible }: IColumn) => visible)
+        .reduce((acc, { width: $width }) => Number(acc) + Number($width), 0),
+    );
+    setRightFixedWidth(
+      rightMappedColumns
+        .filter(({ visible }: IColumn) => visible)
+        .reduce((acc, { width: $width }) => Number(acc) + Number($width), 0),
+    );
+  }, [leftMappedColumns, rightMappedColumns]);
+
+  const onChangeColumns = useCallback((cols, gridPosition) => {
+    if (gridPosition === 'left') {
+      onChangeColumnsFromProps([
+        ...cols,
+        ...centerMappedColumns,
+        ...rightMappedColumns,
+      ]);
+    } else if (gridPosition === 'center') {
+      onChangeColumnsFromProps([
+        ...leftMappedColumns,
+        ...cols,
+        ...rightMappedColumns,
+      ]);
+    } else {
+      onChangeColumnsFromProps([
+        ...leftMappedColumns,
+        ...centerMappedColumns,
+        ...cols,
+      ]);
+    }
+  }, [
+    leftMappedColumns,
+    centerMappedColumns,
+    rightMappedColumns,
+  ]);
 
-  useEffect(() => {
-    if (gridRef.current) gridRef.current.recomputeGridSize();
-  }, [selectedRows]);
 
-  const {
-    header: { height: headerHeight = 0 } = {},
-    totals: { height: totalsHeight = 0 } = {},
-  } = themeRef.current;
+  if (isMultiGrid) {
+    const styles: CSSProperties = {
+      display: 'flex',
+    };
 
+    const multiGridProps = {
+      shouldMovingColumns,
+      shouldChangeColumnsWidth,
+      shouldChangeLeftColumnsWidth,
+      shouldChangeRightColumnsWidth,
+      width: 0,
```

Comment:
If you replace it with width: width || 0, you'll avoid eslint-disable @typescript-eslint/no-non-null-assertion in code below.
Or i missed something?